### PR TITLE
Resource operation with request and response type

### PR DESCRIPTION
**Pull request by chunyu3**
# Cadl Ranch Contribution Checklist:

- [x] I have written a [scenario spec](../docs/writing-scenario-spec.md)
- [x] I have **meaningful** `@scenario` names. Someone can look at the list of scenarios and understand what I'm covering.
- [x] I have written a [mock API](../docs/writing-mock-apis.md)
- [x] I have used `@scenarioDoc`s for extra scenario description and to tell people **how to pass** my mock api check.

### CODE REVIEW

File: packages/cadl-ranch-specs/http/azure/core/basic/main.tsp [hunk 1]
```diff
@@ -354,3 +354,39 @@ Expected response body:
...
+```json
+{
+  "id": 1,
+  "name": "Madge",
+  "nickname": "nickname"
+}
+""")
+@createsResource(User)
+@action("create")
+op createUser is global.Azure.Core.Foundations.ResourceCollectionOperation<User, CreateUserRequest, CreateUserResponse>;
```
Comment (weidongxu-microsoft) [inline packages/cadl-ranch-specs/http/azure/core/basic/main.tsp:392]: service does not suppose to use operation template from `Azure.Core.Foundations`?

**Comment by weidongxu-microsoft** [comment]
I won't block this, but I also won't approve this.

https://github.com/Han-msft/azure-rest-api-specs-pr/pull/1/files#r1378296263

File: packages/cadl-ranch-specs/cadl-ranch-summary.md [hunk 2]
```diff
@@ -190,6 +190,30 @@ Expected response body:
 }
 ```
 
+### Azure_Core_Basic_createUser
```
Comment (timotheeguerin) [inline packages/cadl-ranch-specs/cadl-ranch-summary.md:193]: what is the goal of this test? what does `createUser` mean as a scenario

**Review by timotheeguerin — CHANGES_REQUESTED** [review]

File: packages/cadl-ranch-specs/cadl-ranch-summary.md (hunk 2 above)
Comment (chunyu3) [reply packages/cadl-ranch-specs/cadl-ranch-summary.md:193]: hello @timotheeguerin This test is to cover the senario: the request body and the response body is different from the resource type.

Comment (timotheeguerin) [reply packages/cadl-ranch-specs/cadl-ranch-summary.md:193]: So what specific things from the azure core library does this scenario test. Seems to me it’s basically just adding wrapper on top of what could just be a pure typespec operation

**Pull request closed without merging**